File: CS/IExternalAppointmentCompareSample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Services.Internal;

namespace IExternalAppointmentCompareSample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            AddCustomppointmentComparer(schedulerControl1);
            this.carSchedulingTableAdapter.Fill(this.carsDBDataSet.CarScheduling);
        }

        void AddCustomppointmentComparer(SchedulerControl serviceProvider) {
            MyAppointmentComparerService comparer = new MyAppointmentComparerService("Subject");
            serviceProvider.Services.AddService(typeof(IExternalAppointmentCompareService), comparer);
        }
    }
}

File: VB/IExternalAppointmentCompareSample/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Services.Internal

Namespace IExternalAppointmentCompareSample
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            AddCustomppointmentComparer(schedulerControl1)
			Me.carSchedulingTableAdapter.Fill(Me.carsDBDataSet.CarScheduling)
		End Sub

		Private Sub AddCustomppointmentComparer(ByVal serviceProvider As SchedulerControl)
			Dim comparer As New MyAppointmentComparerService("Subject")
			serviceProvider.Services.AddService(GetType(IExternalAppointmentCompareService), comparer)
		End Sub
	End Class
End Namespace
